### PR TITLE
Check usage for users with active/cancelled/past due subscriptions

### PR DESCRIPTION
**Pull request by aerosol**
So that over limit e-mails and grace period is handled regardless - they're still subscribers.

### Changes

Please describe the changes made in the pull request here.

Below you'll find a checklist. For each item on the list, check one option and delete the other.

### Tests
- [x] Automated tests have been added
- [ ] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [x] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [x] This change does not need a documentation update

### Dark mode
- [ ] The UI has been tested both in dark and light mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/workers/check_usage.ex [hunk 1]
```diff
@@ -43,7 +43,12 @@ defmodule Plausible.Workers.CheckUsage do
           on: s.user_id == u.id,
           left_join: ep in Plausible.Billing.EnterprisePlan,
           on: ep.user_id == u.id,
-          where: s.status == ^Subscription.Status.active(),
+          where:
+            s.status in [
+              ^Subscription.Status.active(),
+              ^Subscription.Status.past_due(),
+              ^Subscription.Status.deleted()
+            ],
```
Comment (zoldar) [inline lib/workers/check_usage.ex:51]: I wonder - shouldn't we pick the most recent subscription per user instead? The way it's done now, won't it evaluate each past subscription in cases where user was switching plans over time?

Comment (aerosol) [reply lib/workers/check_usage.ex:51]: omg so right, I'll see what the implications are tomorrow

Comment (aerosol) [reply lib/workers/check_usage.ex]: It won't evaluate _each_ past subscription, but may make decision (e.g. whether to send an email or not) based on the wrong one (e.g. cancelled). I've fixed the query via 9ddc7bdcc but I still need a test that will fail without it

Comment (aerosol) [reply lib/workers/check_usage.ex]: Pushed [bead241](https://github.com/plausible/analytics/pull/4607/commits/bead241a4ce9c70a09446c2f957e353f21f872f0) and [c630119](https://github.com/plausible/analytics/pull/4607/commits/c630119af2fa2a4abcf2fe5aef62763dd6b6242a) - WYT?

File: lib/workers/check_usage.ex [hunk 2]
```diff
@@ -36,19 +36,31 @@ defmodule Plausible.Workers.CheckUsage do
...
+          least(day_of_month(s.last_bill_date), day_of_month(last_day_of_month(^yesterday))) ==
+            day_of_month(^yesterday),
+        limit: 1
+      )
+
     active_subscribers =
       Repo.all(
         from(u in User,
-          join: s in Plausible.Billing.Subscription,
+          join: s in subquery(last_subscription_query),
```
[the remark this answers is not in the document]
Comment (aerosol) [reply lib/workers/check_usage.ex]: https://github.com/plausible/analytics/pull/4607/commits/1602e3ad6918b881bce97bd565991ac22dea9648

**Review by zoldar — APPROVED** [review]
✨

**Pull request merged**